Automattic/woocommerce-payments
# Avoid crash when seeing the details of an empty deposit

Reported by a merchant.

Sometimes, Stripe will hold some "Reserve funds" on the merchant's balance if it deems the merchant risky. We aren't surfacing those "Reserve funds" anywhere in the UI, they don't appear as Transactions.

If some funds were held in reserve and they're released on the next day, that day's deposit may not have any transactions in it (apart from the released funds). Thus, it's possible to have a Deposit with no transactions in the UI. That resulted in a crash. This PR fixes that crash, although the UI can still be a bit confusing (the big numbers in the Deposit details header don't match the amounts in the list of transactions).

### To test

Apply this patch. It will simulate a deposit with no transactions:
```
diff --git a/includes/admin/class-wc-rest-payments-transactions-controller.php b/includes/admin/class-wc-rest-payments-transactions-controller.php
index 93f0d56d..602e70b9 100644
--- a/includes/admin/class-wc-rest-payments-transactions-controller.php
+++ b/includes/admin/class-wc-rest-payments-transactions-controller.php
@@ -67,6 +67,7 @@ class WC_REST_Payments_Transactions_Controller extends WC_Payments_REST_Controll
 	 * @param WP_REST_Request $request Full data about the request.
 	 */
 	public function get_transactions( $request ) {
+		return [];
 		$page       = (int) $request->get_param( 'page' );
 		$page_size  = (int) $request->get_param( 'pagesize' );
 		$sort       = $request->get_param( 'sort' );
@@ -92,6 +93,7 @@ class WC_REST_Payments_Transactions_Controller extends WC_Payments_REST_Controll
 	 * @param WP_REST_Request $request Full data about the request.
 	 */
 	public function get_transactions_summary( $request ) {
+		return [ 'count' => 0, 'currency' => null, 'store_currencies' => [ 'usd' ], 'total' => 0, 'fees' => 0, 'net' => 0 ];
 		$deposit_id = $request->get_param( 'deposit_id' );
 		$filters    = $this->get_transactions_filters( $request );
 		return $this->forward_request( 'get_transactions_summary', [ $filters, $deposit_id ] );

```

Go to see a deposit's details. On `develop`, the page will crash. On this branch, it will not.
